lib/diagnostic.js

```diff
@@ -5,17 +5,28 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (previous, current) {
+  const workingArray = previous.slice();
+  let accumulator = current;
+  if (typeof current === 'undefined') {
+    accumulator = workingArray.shift();
+  }
+  workingArray.forEach(function (element, index) {
+    accumulator = current (previous, element, index);
+  });
+  return previous;
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
-const sum = function (/* previous, current */) {
-
+const sum = function (previous, current) {
+  for (let index = 0; index < previous.length; index++) {
```

Comment:
A similar problem occurs here.  You're trying to get the ```length``` property of ```previous```, which is assumed to be a number.